<launch>
  <arg name="map_path" default="$(find points_map_loader)/pcd/pointcloud_map.pcd"/>
  <arg name="rviz" default="true"/>
  <arg name="min_range" default="0.5"/>
  <arg name="max_range" default="60.0"/>
  <arg name="downsample_leaf_size" default="2.0"/>
  <arg name="transformation_epsilon" default="0.01"/>
  <arg name="max_iteration" default="30"/>
  <arg name="omp_num_thread" default="4"/>
  <arg name="map_frame_id" default="map"/>
  <arg name="base_frame_id" default="base_link"/>
  <arg name="registration_type" default="FAST_GICP"/>
  <arg name="gicp_num_thread" default="3"/>

  <include file="$(find points_map_loader)/launch/points_map_loader.launch">
    <arg name="map_path" default="$(arg map_path)"/>
  </include>

  <!--node pkg="tf" type="static_transform_publisher" name="velodyne_transform_publisher" args="0.0 0 0.0 0 0 0 /base_link /velodyne 10" /-->

  <node pkg="icp_localization" type="icp_localization_node" name="icp_localization_node">
    <remap from="~points_map" to="points_map"/>
    <remap from="~points_raw" to="points_raw"/>
    <param name="min_range" value="$(arg min_range)"/>
    <param name="max_range" value="$(arg max_range)"/>
    <param name="downsample_leaf_size" value="$(arg downsample_leaf_size)"/>
    <param name="transformation_epsilon" value="$(arg transformation_epsilon)"/>
    <param name="max_iteration" value="$(arg max_iteration)"/>
    <param name="omp_num_thread" value="$(arg omp_num_thread)"/>
    <param name="map_frame_id" value="$(arg map_frame_id)"/>
    <param name="base_frame_id" value="$(arg base_frame_id)"/>
    <param name="registration_type" value="$(arg registration_type)"/>
    <param name="gicp_num_thread" value="$(arg gicp_num_thread)"/>
  </node>

  <group if="$(arg rviz)">
    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find icp_localization)/rviz/icp.rviz"/>
  </group>
</launch>
